src/database/queries/sessions.js

```diff
@@ -748,11 +758,12 @@ exports.deactivateAndReturnMentorSessions = async (userId) => {
 		const foundSessionOwnerships = await SessionOwnership.findAll({
 			attributes: ['session_id'],
 			where: {
-				mentor_id: userId,
+				user_id: userId,
 			},
 			raw: true,
 		})
-		const sessionIds = foundSessionOwnerships.map((ownership) => ownership.session_id)
+		const sessionIdsSet = new Set(foundSessionOwnerships.map((ownership) => ownership.session_id))
```

Comment:
Move this logic the query function itself and then handle it with the flag as we discussed. Let me know if there are any issues with that logic.